insights/core/spec_factory.py

```diff
@@ -268,24 +270,11 @@ class TextFileProvider(FileProvider):
     lines. Each line is filtered if filters are defined for the datasource.
     """
 
-    def create_args(self):
-        args = []
-        filters = "\n".join(get_filters(self.ds)) if self.ds else None
-        if filters:
-            args.append(["grep", "-F", filters, self.path])
-
-        return args
-
     def load(self):
         self.loaded = True
-        args = self.create_args()
-        if args:
-            rc, out = self.ctx.shell_out(args, keep_rc=True, env=SAFE_ENV)
-            self.rc = rc
-            return out
 
         with safe_open(self.path, "r", encoding="utf-8", errors="surrogateescape") as f:
-            return [l.rstrip("\n") for l in f]
+            return [l.rstrip("\n") for l in apply_filters(self.ds, f)]
```

Comment:
Seeing that we are going to use `apply_filters()` to replace the `["grep", "-F", filters, self.path]` in both `TextFileProvider` and `CommandOutputProvider`. 

Just the docstrings of  func `apply_filters` mentions that: 
> Applys filters to the lines of a datasource. This function is used only in
    integration tests. Filters are applied in an equivalent but more performant
    way at run time.

So I am wondering if there could some efficiency loss.  

Any reason for this change?

Reply:
Discussed with @xiangce , this filter method related change was mentioned in RHINENG-7854  . Then this LGTM.

Reply:
I tested the `grep -F` will be faster than `apply_filters` line by line. so I reverted it in the latest commit. Thanks.